app/processImage.js

```diff
@@ -46,6 +48,21 @@ function getCropFunction(cropType) {
     return functionMapping[cropType.toLowerCase()]
 }
 
+function validateImageCropSize(image, size, callback) {
+    if !(size.height && size.width
+    sharp(image)
+        .metadata()
+        .then(function(metadata) {
+            const asp_ratio = metadata.width/metadata.height;
+            if (!size.height) {
+                size.height = Math.round(size.width/asp_ratio);
+            }
+            if (!size.width) {
+                size.width = Math.round(size.height * asp_ratio);
+            }
+            callback(null, image, size)
+        })
```

Comment:
currently this then handles only the success part, It doesn't handle error part. Check https://github.com/harshal27/image-resize-lambda/blob/ae7f762b528f745f673a3a964fbe84a8632fa10c/app/processImage.js#L74 for reference.